src/Metadata/PropertyTypeIterable.php

```diff
@@ -4,67 +4,108 @@
+ *  - and the traversable classes of each are either not present on either sides, or are the same, or parent-child of one another
  */
-final class PropertyTypeIterable extends PropertyTypeArray
+final class PropertyTypeIterable extends AbstractPropertyType
 {
+    /**
+     * @var PropertyType
+     */
+    private $subType;
+
+    /**
+     * @var bool
+     */
+    private $hashmap;
+
     /**
      * @var string
      */
-    private $collectionClass;
+    private $traversableClass;
 
     /**
-     * @param class-string<\Traversable>|null $collectionClass
+     * @param class-string<\Traversable>|null $traversableClass
      */
-    public function __construct(PropertyType $subType, bool $hashmap, bool $nullable, string $collectionClass = null)
+    public function __construct(PropertyType $subType, bool $hashmap, bool $nullable, string $traversableClass = null)
     {
-        parent::__construct($subType, $hashmap, $nullable, null != $collectionClass);
+        parent::__construct($nullable);
 
-        $this->collectionClass = $collectionClass;
+        $this->subType = $subType;
+        $this->hashmap = $hashmap;
+        $this->traversableClass = $traversableClass;
     }
 
     public function __toString(): string
     {
         if ($this->subType instanceof PropertyTypeUnknown) {
-            return 'array'.($this->isCollection() ? '|\\'.$this->collectionClass : '');
+            return 'array'.($this->hasTraversableClass() ? '|\\'.$this->traversableClass : '');
         }
 
         $array = $this->isHashmap() ? '[string]' : '[]';
-        if ($this->isCollection()) {
+        if ($this->hasTraversableClass()) {
             $collectionType = $this->isHashmap() ? ', string' : '';
-            $array .= sprintf('|\\%s<%s%s>', $this->collectionClass, $this->subType, $collectionType);
+            $array .= sprintf('|\\%s<%s%s>', $this->traversableClass, $this->subType, $collectionType);
         }
 
-        return ((string) $this->subType).$array.AbstractPropertyType::__toString();
+        return ((string) $this->subType).$array.parent::__toString();
     }
 
-    public function getCollectionClass(): ?string
+    public function isHashmap(): bool
     {
-        return $this->collectionClass;
+        return $this->hashmap;
     }
 
-    public function isCollection(): bool
+    /**
+     * Returns the type of the next level, which could be an array or hashmap or another type.
+     */
+    public function getSubType(): PropertyType
     {
-        return null != $this->getCollectionClass();
+        return $this->subType;
+    }
+
+    public function getTraversableClass(): string
+    {
+        if (!$this->hasTraversableClass()) {
+            throw new \UnexpectedValueException("Iterable type '{$this}' is not traversable.");
+        }
+
+        return $this->traversableClass;
+    }
+
+    public function hasTraversableClass(): bool
```

Comment:
why not like the previous method call this `isTraversable`?

Reply:
and we need to add this method to the 1.x branch as well (once we decided on the name) so that users can support the newer 1.x branch as well as 2.x in parallel, and we can then deprecated isCollection already in the 1.x branch.